### PR TITLE
MTDSA-3558 - Route based on accept header

### PR DESCRIPTION
**Pull request by jeremystone**

### CODE REVIEW

File: app/routing/VersionRoutingRequestHandler.scala [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      case Some(version) =>
+        versionRoutingMap.versionRouter(version) match {
+          case Some(versionRouter) if featureSwitch.isVersionEnabled(version) => routeWith(versionRouter)(request)
+          case Some(_) => Some(unsupportedVersionAction)
+          case None => Some(unsupportedVersionAction)
+        }
+      case None => Some(invalidAcceptHeaderError)
+    }
+
+    documentHandler orElse apiHandler
```
Comment (MarkAKelly) [inline app/routing/VersionRoutingRequestHandler.scala]: This is nice :+1:

File: app/config/FeatureSwitch.scala [hunk 2]
```diff
@@ -19,13 +19,15 @@ package config
 import play.api.Configuration
 
 case class FeatureSwitch(value: Option[Configuration]) {
-  val DEFAULT_VALUE = true
 
+  private val versionRegex = """(\d)\.\d""".r
+
+  val DEFAULT_VALUE = true
```
Comment (MarkAKelly) [inline app/config/FeatureSwitch.scala]: Is this used?

File: test/routing/VersionRoutingRequestHandlerSpec.scala [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+  "Routing requests with no version" should {
+    implicit val acceptHeader: None.type = None
+
+    handleWithDefaultRoutes(defaultRouter)
+  }
+
+  "Routing requests with valid version" should {
+    implicit val acceptHeader: Some[String] = Some("application/vnd.hmrc.1.0+json")
+
+    handleWithDefaultRoutes(defaultRouter)
```
Comment (MarkAKelly) [inline test/routing/VersionRoutingRequestHandlerSpec.scala]: Should this test go to the non-default router?

File: app/config/FeatureSwitch.scala (hunk 2 above)
Comment (narendravijayarao) [reply app/config/FeatureSwitch.scala]: no usages... :)

File: test/routing/VersionRoutingRequestHandlerSpec.scala (hunk 3 above)
Comment (narendravijayarao) [reply test/routing/VersionRoutingRequestHandlerSpec.scala]: Yes..

File: app/config/FeatureSwitch.scala [hunk 4]
```diff
@@ -38,14 +38,30 @@ case class FeatureSwitch(value: Option[Configuration]) {
...
+      enabled   <- config.getOptional[Boolean](s"version-$versionNo.enabled")
+    } yield enabled
+
+    enabled.getOrElse(false)
+  }
 }
 
 sealed case class FeatureConfig(config: Configuration) {
 
   def isSummaryEnabled(source: String, summary: String): Boolean = {
```
Comment (MarkAKelly) [inline app/config/FeatureSwitch.scala]: What is this?

Comment (narendravijayarao) [reply app/config/FeatureSwitch.scala]: Not sure... it is there in all our services... so didn't touch...

Comment (narendravijayarao) [reply app/config/FeatureSwitch.scala]: but never used in our projects...

**Pull request merged**